### ISSUE TITLE
how to return NDCG as a metric

### ISSUE THREAD

**Issue by Lulu20220** [tensorflow/recommenders#687]
Hello,

I am following the tutorial here https://www.tensorflow.org/recommenders/examples/dcn to build a dcn model, but I want to have ndcg as a metric. Then I did 
`model.compile(optimizer='sgd', metrics=[tfr.keras.metrics.NDCGMetric()])
`

but still not returning ndcg.

Could someone guide me through here please?

**Comment by patrickorlando** [comment]
@Lulu20220 The custom model defined in the tutorial overrides the default keras train and test steps so passing a metric to `compile` will not work.
You need to pass the metric object to the the `Ranking` task defined in the `DCN` model [here](https://www.tensorflow.org/recommenders/examples/dcn#model_construction_2).
